kata02/chop_test.go

```diff
@@ -27,12 +28,22 @@ func TestChop1(t *testing.T) {
 		{7, medHaystack, 3},
 		{0, medHaystack, -1},
 		{8, medHaystack, -1},
+		{7, largeHaystack, 3},
+		{19, largeHaystack, 9},
+		{35, largeHaystack, 18},
+		{99, largeHaystack, -1},
+		{101, largeHaystack, -1},
 	}
 
-	for _, table := range tables {
-		result := Chop1(table.needle, table.haystack)
-		if result != table.result {
-			t.Errorf("Chop1 of (%v) in (%v) was incorrect, got: %v, want: %v.", table.needle, table.haystack, result, table.result)
+	chops := []Chop{Chop1, Chop2}
+
+	for i, chop := range chops {
+		for _, table := range tables {
+			result := chop(table.needle, table.haystack)
+			if result != table.result {
+				t.Errorf("Chop%d of (%v) in (%v) was incorrect, got: %v, want: %v.", i+1, table.needle, table.haystack, result, table.result)
```

Comment:
is there a good way to get the name of the chop method called for the Error string?  instead of indexing on the chop range?  like `typeof(chop)` or something?

Reply:
Its not pretty, but https://stackoverflow.com/questions/7052693/how-to-get-the-name-of-a-function-in-go

Reply:
ha, I saw that stackoverflow, and was like "there must be a better way". I guess not.

Reply:
If you were making something with performance in mind, you would avoid using reflect as much as possible.